core/src/main/java/org/apache/gravitino/utils/MetadataObjectUtil.java

```diff
@@ -98,4 +104,77 @@ public static NameIdentifier toEntityIdent(String metalakeName, MetadataObject m
             "Unknown metadata object type: " + metadataObject.type());
     }
   }
+
+  /**
+   * This method will check if the entity is existed explicitly, internally this check will load the
+   * entity from underlying sources to entity store if not stored, and will allocate an uid for this
+   * entity, with this uid tags can be associated with this entity. This method should be called out
+   * of the tree lock, otherwise it will cause deadlock.
+   *
+   * @param metalake The metalake name
+   * @param object The metadata object
+   * @param env The Gravitino environment
+   * @throws NoSuchMetadataObjectException if the metadata object type doesn't exist.
+   */
+  public static void checkMetadataObject(String metalake, MetadataObject object, GravitinoEnv env) {
+    NameIdentifier identifier = toEntityIdent(metalake, object);
```

Comment:
I think we doesn't need `GravitinoEnv env` params in the `checkMetadataObject(...)` function.
We can directly call `GravitinoEnv.getInstance()` in the function body.

Reply:
Using `env` as method parameter will make the mock test easy to achieve.